File: Revised/Jav/WordCount/Reducer.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class Reducer {

    public static void main(String[] args) throws IOException {
        // Input file path
        String inputFile = "mapper_output.txt";

        // Read input file
        BufferedReader reader = new BufferedReader(new FileReader(inputFile));
        String line;
        StringBuilder content = new StringBuilder();
        while ((line = reader.readLine()) != null) {
            content.append(line).append("\n");
        }
        reader.close();

        // Split the input into words
        String[] words = content.toString().split("\\s+");

        // Map phase
        Map<String, Integer> wordCountMap = new HashMap<>();
        for (String word : words) {
            // Skip counting if the word consists entirely of digits
            if (word.matches("\\d+")) {
                continue;
            }
            
            // Update word count
            if (wordCountMap.containsKey(word)) {
                wordCountMap.put(word, wordCountMap.get(word) + 1);
            } else {
                wordCountMap.put(word, 1);
            }
        }

        // Reduce phase
        for (Map.Entry<String, Integer> entry : wordCountMap.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }
    }
}

File: Revised/Py/WordCount/Reducer.py
prev_word = None
prev_count = 0

input_file_path = "mapper_output.txt"

with open(input_file_path, "r") as input_file:
    for line in input_file:
        line = line.strip()
        word, count = line.split('\t')
        count = int(count)

        if prev_word == word:
            prev_count += count
        else:
            if prev_word:
                print('%s\t%s' % (prev_word, prev_count))
            prev_count = count
            prev_word = word

    # Print the final word count outside of the loop
    if prev_word:
        print('%s\t%s' % (prev_word, prev_count))

